proto/buf/alpha/breaking/v1/config.proto

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020-2021 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.breaking.v1;
+
+// Config represents the breaking change configuration for a module. The rule IDs and categories are defined
+// by the rules_version and apply across the config. The rules_version is independent of the version of
+// the package. The package version refers to the config shape, the rules_version indicates which
+// rule IDs and catetgories should be used.
+//
+// The rule IDs and categories are not encoded as enums in this package because we may want to support custom rule
+// IDs and categories in the future. Callers will need to resolve the rule IDs and categories strings.
+message Config {
+  // rules_version represents the version of the breaking change rule IDs and categories that should be used with this config.
+  string rules_version = 1;
+  // use_rule_ids lists the rule IDs and/or categories that are included in the breaking change check.
+  repeated string use_rule_ids = 2;
+  // except_rule_ids lists the rule IDs and/or categories that are excluded from the breaking change check.
+  repeated string except_rule_ids = 3;
+  // ignore_paths lists the paths of directories and/or files that should be ignored by the breaking change check.
+  // All paths are relative to the root of the module.
+  repeated string ignore_paths = 4;
+  // ignore_rule_ids_paths is a map of rule IDs and/or categories to directory and/or file paths to exclude from the
+  // breaking change check.
+  repeated RuleIDsPaths ignore_rule_ids_paths = 5;
+  // ignore_unstable_packages ignores packages with a last component that is one of the unstable forms recognised
+  // by the PACKAGE_VERSION_SUFFIX:
+  //   v\d+test.*
+  //   v\d+(alpha|beta)\d+
+  //   v\d+p\d+(alpha|beta)\d+
+  bool ignore_unstable_packages = 6;
+}
+
+// RuleIDsPaths represents a rule ID or category and the file and/or directory paths that are ignored for the rule.
+message RuleIDsPaths {
```

Comment:
Since we only encode a single `rule_id`, we might want this to be `RuleIDPaths`. Plus, we might actually want it to be `IDPaths` like @bufdev suggested since it's technically a RuleID or a CategoryID.